<H1
    >Palmer Penguin Data Analytics


Analyst: Hufana

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook </p>
<p>Converting dataset to to dataframe </p>
<ul><strong>Requirements</strong>
    <li>name of the dataframe </li>
    <li>Source </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets//penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3> Descriptive Analytics </h3>

In [4]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3> Cleaning the Data Frame </h3>

In [5]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3> Syntax </h3>

<h3> data_frame['column_name']

In [6]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [7]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [8]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [9]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [10]:
penguins['species'] == "Chinstrap"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [11]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [12]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [14]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [15]:
(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [16]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [17]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [18]:
chinstrap_heavy = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

In [19]:
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1> HOW TO FILL UP NULL VALUES </h1>

In [20]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [21]:
penguins['bill_length_mm']

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [22]:
penguins['bill_length_mm'].mean()

43.9219298245614

In [23]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [24]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [25]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [27]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [28]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [29]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<h1> How to Change the Data Type of a Column </h1>

In [30]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [31]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [33]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<h1> How to create a column </h1>

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [35]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [36]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [37]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1> Exploratory Data Analytics (EDA) </h1>

<h3> Which species has the heaviest body mass in kg? </h3>

In [38]:
chinstrap_df['body_mass_g'].mean()

3733.0882352941176

In [39]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [40]:
adelie_df['body_mass_g'].mean()

3703.95891043398

In [41]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [42]:
gentoo_df['body_mass_g'].mean()

5068.965761177136

<h1> Insight # </h1>

<h3> 1. The species with the heaviest body mass is the Gentoo </h3>

<h3> Which species has the longest bill length in mm? </h3>

In [43]:
chinstrap_df['bill_length_mm'].mean()

48.83382352941177

In [44]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [45]:
adelie_df['bill_length_mm'].mean()

38.82514427516159

In [46]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [47]:
gentoo_df['bill_length_mm'].mean()

47.475983305036785

<h1> Insight #1 </h1>

<h3> The species who have the longest bill length is the Chinstrap species. </h3>

<h3> 2. What are the minimum and maximum bill lengths in the dataset? </h3>

In [48]:
penguins['bill_length_mm'].min(), penguins['bill_length_mm'].max()


(32.1, 59.6)

<h1> Insight #2 </h1>

<h3> The minimum and maximum bill lengths is 32.1 and 59.6 </h3>

<h3> 3. Which of the two species has the heaviest mass in kg? </h3>

In [49]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [50]:
adelie_df['mass_kg'].mean()

3.7052631578947373

In [51]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [52]:
gentoo_df['mass_kg'].mean()

5.069354838709677

<h1> Insight #3 </h1>

<h3> The species who have the heaviest mass is the Gentoo </h3>

<h3> 4. How many penguins are in the each species dataset? </h3>

In [53]:
len(penguins)

344

<h1> Insight #4 </h1>

<h3> The total number or penguins in the dataset os 344 </h3>

<h3> 5. How many penguins are there for each species? </h3>

In [54]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1> Insight #5 </h1>

<h3> The total of penguins in each island is: Adelie 152, Gentoo 124 and on Chinstrap 68. </h3>

<h3> 6. What is the average body mass per sex? </h3>

In [55]:
penguins.groupby('sex')['body_mass_g'].mean()

sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h1> Insight #6 </h1>

<h3> The average body mass of female is 3862.27 and the male is 4514.68 </h3>

<h3> 7. How many penguins are on each island? </h3>

In [56]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1> Insight #7 </h1>

<h3> The total of penguins in each island is: Biscoe       168, Dream        124 and in Torgersen     52. </h3>

<h3> 8. Which species has the longest flipper length in mm? </h3>

In [78]:
chinstrap_df['flipper_length_mm'].mean()

195.8235294117647

In [79]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [80]:
adelie_df['flipper_length_mm'].mean()

190.02575792551556

In [81]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [82]:
gentoo_df['flipper_length_mm'].mean()

217.05576777966422

<h1> Insight #8 </h1>

<h3> The species who has the longest flipper length in mm is the Gentoo species </h3>

<h3> 9. Which species has the most male? </h3>

In [84]:
PerIsland = penguins.groupby(['species', 'island', 'sex']).size()

In [85]:
PerIsland

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h1> Insight #9 </h1>

<h3> The species who have the most male is in Gentoo </h3>

<h3> 10. Which species has the most female? </h3>

In [86]:
PerIsland = penguins.groupby(['species', 'island', 'sex']).size()

In [87]:
PerIsland

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h1> Insight #10 </h1>

<h3> The species who have the most female is in Gentoo </h3>

<h3> 11. Which Island has the least female? </h3>

In [88]:
PerIsland = penguins.groupby(['species', 'island', 'sex']).size()

In [89]:
PerIsland

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h1> Insight #11 </h1>

<h3> The island who have the least female is Biscoe </h3>

<h3> 12. What is the correlation between the bill depth and body mass of the Adelie species? </h3>

In [95]:
adelie_p = penguins[penguins['species'] == "Adelie"]

In [97]:
adelie_Billmass = adelie_p['bill_depth_mm'].corr(adelie_p['body_mass_g'])
adelie_Billmass

0.5649895160254269

<h1> Insight #12 </h1>

<h3> This indicates the correlation between the bill depth and body mass of the Adelie Species </h3>

<h1> Data Aggregation </h1>

<h3> Ilan species per island ang nakatira? </h3>

In [57]:
PerIsland = penguins.groupby(['species', 'island']).count()

In [58]:
PerIsland

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [59]:
PerIsland = penguins.groupby(['species', 'island', 'sex', 'year']).size()

In [60]:
PerIsland

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Biscoe     female  2007     0
                              2008     0
                              2009     0
                      male    2007     0
                              2008     0
                      

In [61]:
avg_species_island = penguins.groupby(['species', 'island'])['body_mass_g'].mean()

In [62]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_g, dtype: float64

In [63]:
avg_species_island = penguins.groupby(['species', 'island', 'sex'])['body_mass_g'].mean()

In [64]:
avg_species_island

species    island     sex   
Adelie     Biscoe     female    3369.318182
                      male      4050.000000
           Dream      female    3344.444444
                      male      4008.620690
           Torgersen  female    3395.833333
                      male      3990.241228
Chinstrap  Biscoe     female            NaN
                      male              NaN
           Dream      female    3527.205882
                      male      3938.970588
           Torgersen  female            NaN
                      male              NaN
Gentoo     Biscoe     female    4679.741379
                      male      5411.011430
           Dream      female            NaN
                      male              NaN
           Torgersen  female            NaN
                      male              NaN
Name: body_mass_g, dtype: float64

In [67]:
mass_species = penguins.groupby(['species'])['body_mass_g'].mean()

In [68]:
mass_species

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h1> TESTING CORRELATION </h1>
<h3> Pearson R Correlation </h3>
<hr>
<h3> Used for testing correlation between two NUMERICAL VALUES </h3>

In [65]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [66]:
weight_flipper_cor

0.8712017673060114

<h1> Numerical - Categorical Correlation Test </h1>
<h3> Library - SciPy (Scientific Python) </h3>
<hr>
<h3> from scipy.stats import f_oneway </h3>

In [69]:
from scipy.stats import f_oneway

In [70]:
penguins['sex']

0        male
1      female
2      female
3        male
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [71]:
x_df = penguins[penguins['sex'] == 'male']

In [72]:
x_df['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [76]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [77]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [73]:
f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

F_onewayResult(statistic=68.38476171974526, pvalue=3.0226600487050253e-15)

In [74]:
result, pvalue = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [75]:
pvalue

3.0226600487050253e-15